packages/core/src/combinator/withArrayValues.js

```diff
@@ -1,15 +1,16 @@
 /** @license MIT License (c) copyright 2010-2017 original author or authors */
 
-import { empty } from '../source/empty'
 import Pipe from '../sink/Pipe'
+import { empty } from '../source/empty'
+import { take } from './slice'
 
 export const withArrayValues = (array, stream) =>
   zipArrayValues(keepLeft, array, stream)
 
 export const zipArrayValues = (f, array, stream) =>
   array.length === 0 || stream === empty()
     ? empty()
-    : new ZipArrayValues(f, array, stream)
+    : new ZipArrayValues(f, array, take(array.length, stream))
```

Comment:
What do you think about adding a test to cover this, i.e., based on the code where you discovered the bug?

Reply:
I can't believe I missed that :flushed: